Calculator/Calculator/Model/Operator.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Operator.swift
+//  Calculator
+//
+//  Created by mireu on 2023/10/06.
+//
+
+enum Operator: Character, CaseIterable, CalculateItem {
+    case add = "+"
+    case substract = "-"
+    case divide = "/"
+    case multiply = "*"
+    
+    func calculate(lhs: Double, rhs: Double) throws -> Double {
+        switch self {
+        case .add:
+            return add(lhs: lhs, rhs: rhs)
+        case .substract:
+            return substract(lhs: lhs, rhs: rhs)
+        case .divide:
+            return try divide(lhs: lhs, rhs: rhs)
+        case .multiply:
+            return multiply(lhs: lhs, rhs: rhs)
+        }
+    }
+    
+    private func add(lhs: Double, rhs: Double) -> Double {
+        return lhs + rhs
+    }
+    
+    private func substract(lhs: Double, rhs: Double) -> Double {
+        return lhs - rhs
+    }
+    
+    private func divide(lhs: Double, rhs: Double) throws -> Double {
+        if rhs == .zero {
+            throw CalculatorError.divideByZero
+        }
+        return lhs / rhs
+    }
```

Comment:
0으로 나눌 경우 적절한 오류처리를 해주셨네요 👍👍👍

Double 타입에는 `nan` 이라는 수가 아님(not a number)을 나타내는 타입 프로퍼티가 있습니다.
이를 활용해보는 방법도 있어요 😊
```suggestion
    private func divide(lhs: Double, rhs: Double) -> Double {
        if rhs == .zero {
            return .nan
        }
        return lhs / rhs
    }
```

오류처리를 하는 방법과 오류처리를 하지 않고 조건문을 활용하여 예외사항을 만드는 방법 모두 괜찮다고 생각합니다.
일반적으로 오류처리가 조금 더 안전할 수도 있겠네요~!
두 방법 중 하나를 선택할 때 자신만의 기준을 세우시길 바랍니다 👍
이 부분은 수정하지 않으셔도 좋습니다.